src/main/java/seedu/address/model/consultation/Consultation.java

```diff
@@ -0,0 +1,104 @@
+package seedu.address.model.consultation;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+import seedu.address.model.student.Student;
+
+/**
+ * Represents a Consultation in the system.
+ * Guarantees: details are present and not null, field values are validated.
+ */
+public class Consultation {
+
+    // Identity fields
+    private final Date date;
+    private final Time time;
+
+    // Data field
+    private final List<Student> students;
+
+    /**
+     * Every field must be present and not null.
+     * The list of students can be empty but not null.
+     */
+    public Consultation(Date date, Time time, List<Student> students) {
+        requireAllNonNull(date, time);
+        this.date = date;
+        this.time = time;
+
+        this.students = students != null ? new ArrayList<>(students) : new ArrayList<>();
+    }
+
+    public Date getDate() {
+        return date;
```

Comment:
Maybe consider adding testcases for these flagged by Codecov - should be trivial (just getters and equality test), and Codecov will be happy.